server/src/main/java/org/folio/reservoir/server/UploadService.java

```diff
@@ -29,41 +30,16 @@ public class UploadService {
   private Future<Void> uploadPayloadStream(ReadStream<JsonObject> upload,
       IngestWriteStream ingestWriteStream) {
     Promise<Void> promise = Promise.promise();
-    if (ingestWriteStream != null) {
-      ingestWriteStream.drainHandler(x -> upload.resume());
-    }
-    AtomicInteger number = new AtomicInteger();
-    AtomicInteger errors = new AtomicInteger();
+    ingestWriteStream.drainHandler(x -> upload.resume());
```

Comment:
You can replace all work the flow between upload (ReadStream) and ingestWriteStream with a Pump.
```suggestion
```